articles/mysql/single-server/how-to-troubleshoot-high-cpu-utilization.md

```diff
@@ -21,6 +21,32 @@ For example, consider a sudden surge in connections that initiates surge of data
 
 Besides capturing metrics, it’s important to also trace the workload to understand if one or more queries are causing the spike in CPU utilization.
 
+## High CPU causes
+
+CPU spikes can occur for various reasons, primarily due to spikes in connections and poorly written SQL queries, or a combination of both:
+
+#### Spike in connections
+
+An increase in connections can lead to an increase in threads, which in turn can cause a rise in CPU usage as it has to manage these connections along with their queries and resources. To troubleshoot a spike in connections, you should check the [Total Connections](./../flexible-server/concepts-monitoring.md#list-of-metrics) metric and refer to the next section for more details about these connections. You can utilize the performance_schema to identify the hosts and users currently connected to the server with the following commands:
+
+Current connected hosts
+```
+   select HOST,CURRENT_CONNECTIONS From performance_schema.hosts
+   where CURRENT_CONNECTIONS > 0
+   and host not in ('NULL','localhost');
+```
+
+Current connected users
+```
+   select USER,CURRENT_CONNECTIONS from performance_schema.users
+   where CURRENT_CONNECTIONS >0
+   and USER not in ('NULL','azure_superuser');
+```
+
+#### Poorly written SQL queries
+
+Queries that are expensive to execute and scan a large number of rows without an index, or those that perform temporary sorts along with other inefficient plans, can lead to CPU spikes. While some queries may execute quickly in a single session, they can cause CPU spikes when run in multiple sessions. Therefore, it’s crucial to always explain your queries that you capture from the [show processlist](https://dev.mysql.com/doc/refman/5.7/en/show-processlist.html) and ensure their execution plans are efficient. This can be achieved by ensuring they scan a minimal number of rows by using filters/where cluase, utilize indexes and avoid using large temporary sort along with other bad execution plans. For more information about execution plans, see [EXPLAIN Output Format](https://dev.mysql.com/doc/refman/5.7/en/explain-output.html).
```

Comment:
```suggestion
Queries that are expensive to execute and scan a large number of rows without an index, or those that perform temporary sorts along with other inefficient plans, can lead to CPU spikes. While some queries may execute quickly in a single session, they can cause CPU spikes when run in multiple sessions. Therefore, it’s crucial to always explain your queries that you capture from the [show processlist](https://dev.mysql.com/doc/refman/5.7/en/show-processlist.html) and ensure their execution plans are efficient. This can be achieved by ensuring they scan a minimal number of rows by using filters/where clause, utilize indexes and avoid using large temporary sort along with other bad execution plans. For more information about execution plans, see [EXPLAIN Output Format](https://dev.mysql.com/doc/refman/5.7/en/explain-output.html).
```